src/control/lib/hardware/pretty.go

```diff
@@ -23,25 +23,29 @@ func PrintTopology(t *Topology, output io.Writer) error {
 		return nil
 	}
 
-	for _, numaNode := range t.NUMANodes {
+	for _, numaNode := range t.NUMANodes.AsSlice() {
 		coreSet := &system.RankSet{}
 		for _, core := range numaNode.Cores {
 			coreSet.Add(system.Rank(core.ID))
 		}
 
 		fmt.Fprintf(ew, "NUMA Node %d\n", numaNode.ID)
 		fmt.Fprintf(ew, "  CPU cores: %s\n", coreSet)
-		fmt.Fprintf(ew, "  PCI devices:\n")
-		for addr, devs := range numaNode.PCIDevices {
-			fmt.Fprintf(ew, "    %s\n", &addr)
-			for _, dev := range devs {
-				fmt.Fprintf(ew, "      %s\n", dev)
-			}
-		}
 		fmt.Fprintf(ew, "  PCI buses:\n")
 		for _, bus := range numaNode.PCIBuses {
 			fmt.Fprintf(ew, "    %s\n", bus)
+			for addr, devs := range numaNode.PCIDevices {
```

Comment:
It occurs to me that we might want to sort the addresses in lexical order for output, too.

Reply:
Done.